### ISSUE TITLE
Issue 04: coarse inference で pickability/usability score を NPZ に保存し physics_input_score を作る

### ISSUE THREAD

**Issue by ozw4** [ozw4/seisai#365]
# Issue 04: coarse inference で pickability/usability score を NPZ に保存し physics_input_score を作る

## 背景

Step 2 の目的は、coarse 終了後の physics で使う pick 観測点を選別することである。そのため、coarse inference の出力 NPZ に trace-level score を保存し、physics 側で `prob_key` として使える `physics_input_score` を追加する。

現状 `run_coarse_infer()` は `coarse_pick_i`, `coarse_pmax`, `coarse_prob_summary` を保存している。multi-task model では `pickability_logits` / `coarse_usable_logits` も出る。

## ゴール

coarse NPZ に以下の optional keys を保存する。

```text
coarse_pickability_score   float32[n_traces], in [0,1]
coarse_usable_score        float32[n_traces], in [0,1], optional Step 3
physics_input_score        float32[n_traces], in [0,1]
```

legacy config では保存しなくてもよいが、`infer.physics_input_score.mode: pmax` の場合は `physics_input_score = coarse_pmax` を保存してもよい。

## 変更対象の候補

- `packages/seisai-engine/src/seisai_engine/pipelines/fbpick/coarse/infer.py`
- `packages/seisai-engine/src/seisai_engine/pipelines/fbpick/common/io.py`
- `packages/seisai-engine/src/seisai_engine/pipelines/fbpick/coarse/output.py`
- `packages/seisai-engine/src/seisai_engine/pipelines/fbpick/coarse/qc.py` または QC writer 呼び出し

### 1. infer で multi-task output を unwrap する

`run_coarse_infer()` 内の

```python
logits = model(x_bchw)
```

を次の流れに変える。

```python
output = model(x_bchw)
pick_logits = get_pick_logits(output)
pickability_logits = get_optional_trace_logits(output, PICKABILITY_LOGITS_KEY)
coarse_usable_logits = get_optional_trace_logits(output, COARSE_USABLE_LOGITS_KEY)
```

以降の coarse pick 抽出は `pick_logits` を使う。

### 2. anchor trace score を full trace へ restore する

現在 `restore_anchor_predictions_to_full_traces()` は pick と pmax を同時に restore している。trace-level score 用に汎用 helper を追加する。

候補:

```python
def restore_anchor_vector_to_full_traces(
    *,
    anchor_values: np.ndarray,
    trace_valid: np.ndarray,
    anchor_source_pos: np.ndarray,
    segment_id: np.ndarray,
    segments,
    n_traces: int,
    value_name: str,
) -> np.ndarray:
    ...
```

要件:

- segment 内で anchor positions に対して線形補間する。
- `coarse_pmax` の restore と同じ segment metadata validation を使う。
- 出力 shape は `(n_traces,)`。
- score は補間後に `[0,1]` に clip して `float32`。

`pickability_logits` は sigmoid してから restore する。

```python
anchor_pickability_score = torch.sigmoid(pickability_logits).cpu().numpy()
```

### 3. channel tile merge を拡張する

現在は `coarse_pick_weighted_sum`, `coarse_pmax_weighted_sum` を tile weights で集約している。同じ方式で score も集約する。

追加候補:

```python
pickability_weighted_sum
coarse_usable_weighted_sum
physics_input_score_weighted_sum
```

`counts` と `coarse_weight_sum` は既存を使う。

### 4. physics_input_score を作る

`infer.physics_input_score.mode` に従って作る。

```python
if mode == "pmax":
    physics_input_score = coarse_pmax
elif mode == "pickability_times_pmax":
    physics_input_score = coarse_pickability_score * coarse_pmax
elif mode == "pickability_times_usability":
    physics_input_score = coarse_pickability_score * coarse_usable_score
elif mode == "pickability_times_usability_times_pmax":
    physics_input_score = coarse_pickability_score * coarse_usable_score * coarse_pmax
```

必要な component が無い mode を指定した場合はエラー。

### 5. save_coarse_npz を拡張する

`common/io.py::save_coarse_npz()` に optional `extra_arrays` か明示的 optional 引数を追加する。

推奨:

```python
extra_arrays: Mapping[str, np.ndarray] | None = None
```

validation:

- key は str。
- value は 1D vector, length `n_traces`。
- `coarse_pickability_score`, `coarse_usable_score`, `physics_input_score` は `float32` に coerce し `[0,1]` を検証。
- 既存 required keys を壊さない。

`load_coarse_npz()` は unknown keys をすでに保持しているため、optional key の shape validation を追加する程度でよい。

### 6. QC optional 表示

既存 coarse QC に confidence plot がある場合、`physics_input_score` を表示できるようにする。ただしこの issue では必須ではない。最低限 NPZ 保存と validation を優先する。

## 受け入れ条件

- legacy model + legacy config で coarse inference が従来通り完走する。
- pickability head 有効 + `mode: pickability_times_pmax` で NPZ に `coarse_pickability_score` と `physics_input_score` が保存される。
- usability head 有効 + `mode: pickability_times_usability` で `coarse_usable_score` と `physics_input_score` が保存される。
- 各 score は `float32[n_traces]` で finite かつ `[0,1]`。
- `coarse_pmax` と `coarse_prob_summary` の既存意味を壊さない。
- 必要な head が無い mode を指定すると明確なエラー。

## テスト

新規テスト候補:

- `tests/test_fbpick_coarse_infer_scores.py`

unit test で十分:

1. `restore_anchor_vector_to_full_traces()` が既存 pmax restore と同じ補間を行う。
2. score mode の計算が正しい。
3. missing component でエラー。
4. `save_coarse_npz(extra_arrays=...)` が optional score を保存し、`load_coarse_npz()` で読める。

## 非ゴール

- physics 側 config の変更は Issue 05。
- PU loss の実装は Issue 03。
- OOF usability label は Issue 06。
